api/v1alpha2/contentlibrary_types.go

```diff
@@ -0,0 +1,360 @@
+)
+
+type BaseContentLibrarySpec struct {
+	// ID describes the unique identifier used to find the library in vCenter.
+	//
+	// Please note this value may differ depending on spec.type:
+	// - Type=ContentLibrary -- ID is a content library UUID.
+	// - Type=Inventory      -- ID is a vSphere folder managed object ID.
+	ID string `json:"id"`
+
+	// +optional
+	// +kubebuilder:default=ContentLibrary
+
+	// Type describes the type of library.
+	//
+	// Defaults to ContentLibrary.
+	Type LibraryType `json:"libraryType,omitempty"`
+
+	// +optional
+	// +kubebuilder:default=FromItemID
+
+	// ResourceNamingStrategy describes the naming strategy for item resources
+	// in this content library.
+	//
+	// This field is immutable.
+	//
+	// Defaults to FromItemID.
+	//
+	// This field is only valid for libraries with spec.type=ContentLibrary.
+	ResourceNamingStrategy ResourceNamingStrategy `json:"resourceNamingStrategy,omitempty"`
+}
+
+// ContentLibrarySpec defines the desired state of a ContentLibrary.
+type ContentLibrarySpec struct {
+	BaseContentLibrarySpec `json:",inline"`
+
+	// +optional
+
+	// StorageClass describes the name of the StorageClass used when publishing
+	// images to this library.
+	//
+	// This field is only valid for libraries with spec.type=Inventory and
+	// spec.allowPublish=true.
+	StorageClass string `json:"storageClass,omitempty"`
+
+	// +optional
+
+	// AllowDelete describes whether or not it is possible to delete items from
+	// this library.
+	//
+	// This field is only valid for libraries with spec.type=ContentLibrary.
+	AllowDelete bool `json:"allowDelete,omitempty"`
+
+	// +optional
+
+	// AllowImport describes whether or not it is possible to import remote
+	// OVA/OVF/ISO content to this library.
+	//
+	// This field is only valid for libraries with spec.type=ContentLibrary.
+	AllowImport bool `json:"allowImport,omitempty"`
+
+	// +optional
+
+	// AllowPublish describes whether or not it is possible to publish new items
+	// to this library.
+	AllowPublish bool `json:"allowPublish,omitempty"`
+}
+
+// ContentLibraryStatus defines the observed state of ContentLibrary.
+type ContentLibraryStatus struct {
+	// +optional
+
+	// Name describes the display name for the library.
+	Name string `json:"name,omitempty"`
+
+	// +optional
+
+	// Description describes a human-readable description for the library.
+	Description string `json:"description,omitempty"`
+
+	// +optional
+
+	// StorageBackings describes the default storage backing for the library.
+	StorageBackings []StorageBacking `json:"storageBacking,omitempty"`
+
+	// +optional
+
+	// Version describes a value that tracks changes to the library's metadata,
+	// such as its name or description.
+	//
+	// This field is only valid for libraries with spec.type=ContentLibrary.
+	Version string `json:"version,omitempty"`
+
+	// +optional
+
+	// PublishInfo describes how the library is published.
+	//
+	// This field is only non-empty for published libraries.
```

Comment:
nit: published libraries -> published content libraries

Reply:
I simplified the term from "content library" to library on purpose to make it more generic. See any issues with that?